core/src/main/java/com/linecorp/armeria/common/RequestContextExporterBuilder.java

```diff
@@ -0,0 +1,250 @@
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.RequestContextExporter.ExportEntry;
+
+import io.netty.util.AsciiString;
+import io.netty.util.AttributeKey;
+
+/**
+ * Builds a new {@link RequestContextExporter}.
+ */
+public final class RequestContextExporterBuilder {
+
+    private static final String PREFIX_ATTRS = "attrs.";
+    private static final String PREFIX_HTTP_REQ_HEADERS = "req.http_headers.";
+    private static final String PREFIX_HTTP_RES_HEADERS = "res.http_headers.";
+
+    private final Set<BuiltInProperty> builtIns = EnumSet.noneOf(BuiltInProperty.class);
+    private final Set<ExportEntry<AttributeKey<?>>> attrs = new HashSet<>();
+    private final Set<ExportEntry<AsciiString>> httpReqHeaders = new HashSet<>();
+    private final Set<ExportEntry<AsciiString>> httpResHeaders = new HashSet<>();
+
+    /**
+     * Adds the specified {@link BuiltInProperty} to the export list.
+     */
+    public RequestContextExporterBuilder addBuiltIn(BuiltInProperty property) {
+        builtIns.add(requireNonNull(property, "property"));
+        return this;
+    }
+
+    /**
+     * Returns {@code true} if the specified {@link BuiltInProperty} is in the export list.
+     */
+    public boolean containsBuiltIn(BuiltInProperty property) {
+        return builtIns.contains(requireNonNull(property, "property"));
+    }
+
+    /**
+     * Returns all {@link BuiltInProperty}s in the export list.
+     */
+    public Set<BuiltInProperty> getBuiltIns() {
+        return ImmutableSet.copyOf(builtIns);
+    }
+
+    /**
+     * Adds the specified {@link AttributeKey} to the export list.
+     *
+     * @param alias the alias of the attribute to export
+     * @param attrKey the key of the attribute to export
+     */
+    public RequestContextExporterBuilder addAttribute(String alias, AttributeKey<?> attrKey) {
+        requireNonNull(alias, "alias");
+        requireNonNull(attrKey, "attrKey");
+        attrs.add(new ExportEntry<>(attrKey, PREFIX_ATTRS + alias, null));
+        return this;
+    }
+
+    /**
+     * Adds the specified {@link AttributeKey} to the export list.
+     *
+     * @param alias the alias of the attribute to export
+     * @param attrKey the key of the attribute to export
+     * @param stringifier the {@link Function} that converts the attribute value into a {@link String}
+     */
+    public RequestContextExporterBuilder addAttribute(String alias, AttributeKey<?> attrKey,
+                                                      Function<?, String> stringifier) {
+        requireNonNull(alias, "alias");
+        requireNonNull(attrKey, "attrKey");
+        requireNonNull(stringifier, "stringifier");
+        attrs.add(new ExportEntry<>(attrKey, PREFIX_ATTRS + alias, stringifier));
+        return this;
+    }
+
+    /**
+     * Returns {@code true} if the specified {@link AttributeKey} is in the export list.
+     */
+    public boolean containsAttribute(AttributeKey<?> key) {
```

Comment:
How about hiding all getters from the public API by making them package-private or removing them? Then, we could add the getters to `RequestContextExporter`, without `get` method prefix.

Reply:
Nice suggestion. I thought that removing getters from the `RequestContextExporterBuilder` too. But if remove them, the getters from `RequestContextExportingAppender` could be breaking changes because the getters could be called while building `RequestContextExporter`.
https://github.com/line/armeria/blob/29befcd911fae48e01761e37638a6473c2e76e30/logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java#L117-L123

I guess the public getters from `RequestContextExportingAppender` could be only used for advanced cases so the breaking changes are not a big impact for most of the users.

Reply:
Had discussed with @trustin, we decide to deprecate the getters from `RequestContextExporterBuilder` and `RequestContextExportingAppender`. https://github.com/line/armeria/pull/2314/commits/49ab82b7d6e060e4c553e39a35d8b46d98d80274